ratex/testing/common.py

```diff
@@ -374,11 +374,58 @@ def run_step(device, model_origin, args, jit_script=True):
     return out
 
 
-def verify_step(model, args, jit_script=True, tol=1e-5):
+def run_step_dx(device, model_origin, args, jit_script=True):
```

Comment:
we should merge run_step and run_step_dx, with an extra argument `with_backward`.
And we should return both loss and args back.